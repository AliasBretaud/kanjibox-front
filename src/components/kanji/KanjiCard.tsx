import { Paper, Typography } from "@mui/material";

import type { $Kanji } from "@/types";
import Table, { Row } from "@/components/ui/Table";

const KanjiCard = ({ kunYomi, onYomi, translations, value }: $Kanji) => (
  <Paper sx={{ p: 2 }} className="w-full max-w-md flex flex-col items-center">
    <Typography
      className="font-kanji"
      sx={{ opacity: 0.75, color: "black", fontSize: 80 }}
    >
      {value}
    </Typography>
    <Table>
      <Row title="音読み" data={onYomi} />
      <Row title="訓読み" data={kunYomi} />
      <Row title="意味" data={translations?.en} />
    </Table>
  </Paper>
);

export default KanjiCard;
