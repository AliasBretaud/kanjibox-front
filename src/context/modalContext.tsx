"use client";

import type { $Kanji } from "@/types";
import React from "react";

export type ModalOptions = Partial<Record<string, string | string[] | $Kanji>>;

type ContextValues =
  | {
      shownModal: string;
      setShownModal: React.Dispatch<React.SetStateAction<string>>;
      modalOptions: ModalOptions;
      setModalOptions: React.Dispatch<React.SetStateAction<ModalOptions>>;
    }
  | undefined;

export const ModalContext = React.createContext<ContextValues>(undefined);

export function ModalProvider({ children }: { children: React.ReactNode }) {
  const [shownModal, setShownModal] = React.useState<string>("");
  const [modalOptions, setModalOptions] = React.useState<ModalOptions>({});
  return (
    <ModalContext.Provider
      value={{ shownModal, setShownModal, modalOptions, setModalOptions }}
    >
      {children}
    </ModalContext.Provider>
  );
}
