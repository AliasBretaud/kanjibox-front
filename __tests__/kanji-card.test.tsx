import KanjiCard from "@/components/kanji/KanjiCard";
import { ModalProvider } from "@/context/modalContext";
import type { $Kanji } from "@/types/models";
import { render, screen } from "@testing-library/react";

const kanjiMock: $Kanji = {
  value: "人",
  onYomi: ["ニン", "ジン"],
  kunYomi: ["ひと"],
  translations: {
    en: ["People", "Person"],
  },
};

describe("KanjiCard component tests", () => {
  it("renders the component", () => {
    render(
      <ModalProvider>
        <KanjiCard {...kanjiMock} locale="en" />
      </ModalProvider>,
    );

    const value = screen.getByText("人");
    const onYomi1 = screen.getByText("ニン");
    const onYomi2 = screen.getByText("ジン");
    const kunYomi = screen.getByText("ひと");
    const translation1 = screen.getByText("People");
    const translation2 = screen.getByText("Person");
    expect(value).toBeInTheDocument();
    expect(onYomi1).toBeInTheDocument();
    expect(onYomi2).toBeInTheDocument();
    expect(kunYomi).toBeInTheDocument();
    expect(translation1).toBeInTheDocument();
    expect(translation2).toBeInTheDocument();
  });

  it("renders homepage unchanged", () => {
    const { container } = render(
      <ModalProvider>
        <KanjiCard {...kanjiMock} locale="en" />
      </ModalProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
