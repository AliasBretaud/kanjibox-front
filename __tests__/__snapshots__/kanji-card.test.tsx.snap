// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KanjiCard component tests renders homepage unchanged 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 w-full max-w-md flex flex-col items-center css-i0bk1x-MuiPaper-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 font-kanji css-1hdi49b-MuiTypography-root"
    >
      人
    </p>
    <table
      class="w-full border-collapse text-left font-kanji border border-solid border-neutral-300"
    >
      <tbody>
        <tr>
          <th
            class="bg-stone-100 text-center text-sm w-[75px] border border-solid border-neutral-300"
            rowspan="2"
          >
            音読み
          </th>
          <td
            class="py-[5px] px-2 border border-solid border-neutral-300"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
            >
              ニン
            </p>
          </td>
        </tr>
        <tr>
          <td
            class="py-[5px] px-2 border border-solid border-neutral-300"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
            >
              ジン
            </p>
          </td>
        </tr>
        <tr>
          <th
            class="bg-stone-100 text-center text-sm w-[75px] border border-solid border-neutral-300"
            rowspan="1"
          >
            訓読み
          </th>
          <td
            class="py-[5px] px-2 border border-solid border-neutral-300"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
            >
              ひと
            </p>
          </td>
        </tr>
        <tr>
          <th
            class="bg-stone-100 text-center text-sm w-[75px] border border-solid border-neutral-300"
            rowspan="2"
          >
            意味
          </th>
          <td
            class="py-[5px] px-2 border border-solid border-neutral-300"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
            >
              People
            </p>
          </td>
        </tr>
        <tr>
          <td
            class="py-[5px] px-2 border border-solid border-neutral-300"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
            >
              Person
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
